aten/src/ATen/native/quantized/cpu/serialization_versions.h

```diff
@@ -0,0 +1,164 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/core/List.h>
+
+#include <ATen/native/quantized/cpu/fbgemm_utils.h>
+#include <ATen/native/quantized/cpu/qnnpack_utils.h>
+
+#include <tuple>
+
+using SerializationType = std::tuple<
+  std::string /*name of the op that is serialized*/,
+  int64_t /*version*/,
+  std::vector<at::Tensor> /*list of all non-optional tensors*/,
+  std::vector<c10::optional<at::Tensor>> /*list of all optional tensors*/,
+  std::vector<double> /*list of all doubles*/,
+  std::vector<int64_t> /*list of all integers*/
+>;
+
+/* Legacy serialization types */
+
+
+/* Convolution prepacked parameters serialization.
+ *
+ * Files that need to be updated if version changes:
+ * - ATen/native/quantized/cpu/fbgemm_utils.cpp
+ * - jit/passes/onnx/unpack_quantized_weights.cpp
+ *
+ * Version 1 (Legacy)
+ * - Fields:
+ *  1. weight
+ *  2. bias
+ *  3. stride x kSpatialDim
+ *  4. padding x kSpatialDim
+ *  5. dilation x kSpatialDim
+ *  6. groups
+ *
+ * Version 2+
+ * - Fields:
+ *  1. name of the op ("conv")
+ *  2. version number (int64_t)
+ *  3. list of all non-optional tensors (vector<Tensor>)
+ *    1. unpacked weight
+ *  4. list of all optional tensors (vector<optional<Tensor>>)
+ *    1. bias
+ *  5. list of all doubles
```

Comment:
why do we have this now if this is empty?

Reply:
Because we have to have the same number of fields going into the future (forward compatibility). Otherwise, we will end up with the same problem if we decide to add doubles to the tuple

Reply:
>  the same number of fields

this seems pretty restrictive.  Do we know the context on why this restriction exists?  Not really for PyTorch, but in general adding new fields in future versions is a pretty common thing to do when versioning things.

Reply:
why do we need forward compatibility? 

and why double? is there plans to add double in next few PRs?